cirq/experiments/xeb_simulation.py

```diff
@@ -120,7 +120,9 @@ def simulate_2q_xeb_circuits(
         # Need an actual object; not np.random or else multiprocessing will
         # fail to pickle the closure object:
         # https://github.com/quantumlib/Cirq/issues/3717
-        simulator = sim.Simulator(seed=np.random.RandomState())
+        simulator = cast(
+            'cirq.SimulatesIntermediateState', sim.Simulator(seed=np.random.RandomState())
+        )
```

Comment:
mypy complains here because `simulator` is `Option[SimulatesIntermediateState]` and it doesn't realize `Simulator` matches that. If anybody knows of a better option here then lmk.

Reply:
It looks like `Simulator` is a `SimulatesIntermediateStateBase`, not a `SimulatesIntermediateState`:
https://github.com/quantumlib/Cirq/blob/4e1844e58ece3f4f7aa342e64520b4cd0b20bbe2/cirq/sim/sparse_simulator.py#L47-L49
https://github.com/quantumlib/Cirq/blob/4e1844e58ece3f4f7aa342e64520b4cd0b20bbe2/cirq/sim/state_vector_simulator.py#L33-L38
Depending on how the function uses `simulator`, you either need to update the type in the function or prohibit non-`SimulatesIntermediateState` from being passed in.

Reply:
Doh! Fixed.